src/workloads/execution/ConstantFoldArithmetic.yml

```diff
@@ -0,0 +1,113 @@
+SchemaVersion: 2018-07-01
+Owner: "@mongodb/query"
+Description: |
+  SERVER-63099 addresses a correctness issue in how we constant fold during multiplication and addition.
+  This workload tests the impact of the constant folding optimization in the aggregation pipeline to better
+  understand the impact of SERVER-63099 and hopefully mitigate the performance regression with other optimizations.
+
+  The first pipeline is the control and should be unaffected by changes to the associativity of multiplication in the optimizer
+  because it only contains constants.
+
+  The three pipelines in this workload that contain a fieldpath should have different performance characteristics depending
+  on the level of associativity in the optimizer.
+
+  - Full associativity (current situation on master):
+    All three pipelines should have similar performance statistics because all 1000 constants are rearranged to be
+    folded into one another, ignoring the order of operations in the original pipeline.
+
+  - No associativity (simple correctness fix):
+    All three pipelines should have similar performance that is significantly worse than full associativity, because
+    each document in the pipeline will require 1000 multiplication operations because no constant folding will happen
+    with $multiply marked as non-associative.
+
+  - Left associativity (implementation of a correct optimization):
+    StartingFieldpath should see no regression from the "no associativity" case.
+    MiddleFieldpath should see a speedup that puts its performance in between "no associativity" and "full associativity".
+    TailingFieldpath should speed up back to its performance in the "full associativity" case.
```

Comment:
Hmm I'm confused on this one, why isn't it more comparable to middle field path?

Reply:
The TailingFieldpath case looks like `[c1, c2, c3, c4, "$v"]` and optimizes to `[c', "$v"]`, which means that only one multiplication is performed per document. In the full associative and commutative case, that same expression would be optimized to `["$v", c']` (commutative constants are arbitrarily put at the end right now) which is also one multiplication per document